sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -228,9 +235,10 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       //      other choice.
       case j @ ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, nonEquiCond,
           _, left, right, hint) =>
+        val hashJoinSupport = hashJoinSupported(leftKeys, rightKeys)
         def createBroadcastHashJoin(onlyLookingAtHint: Boolean) = {
-          val buildSide = getBroadcastBuildSide(
-            left, right, joinType, hint, onlyLookingAtHint, conf)
+          val buildSide = Option(hashJoinSupport).filter(identity).flatMap(
```

Comment:
I feel it's clearer to do
```
val buildSide = if (hashJoinSupport) {
  getBroadcastBuildSide...
} else {
  None
}
```

Reply:
Sure, this was my initial version as well. I thought that `Option` trick is more in scala style. Reverting back to if/else.